src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -1551,6 +1568,8 @@ mongoc_uri_parse_before_slash (mongoc_uri_t *uri,
       hosts = before_slash;
    }
 
+   mongoc_uri_check_for_genuine_hosts (hosts);
```

Comment:
Related to my response in https://github.com/mongodb/mongo-c-driver/pull/1374#discussion_r1291326865, [DRIVERS-2583](https://jira.mongodb.org/browse/DRIVERS-2583) is unclear about all of the edge cases that must be considered.

@alexbevi: Can you provide some guidance here?

 * Is it sufficient to just match against the initial connection string or should we defer this logic until _after_ all hosts have been parsed (and SRV  initially resolved)?
 * If we defer and check hosts individually, do you agree it only makes sense to log a single message to avoid polluting the logs?
 * Furthermore, can we avoid subsequent checks after SRV polling?

Reply:
Pushed changes. Detection is now done in `mongoc_topology_new()` and will take place _after_ the initial SRV lookup. It's not feasible to mock initial SRV responses in libmongoc, so that is intentionally left untested.

I also switched to using a rear-anchored host name match (consistent with mongosh) and added test coverage for that.

And we only log a single message when the first nongenuine host is detected. That behavior is unchanged from the first draft of this PR.

Reply:
Revised the PR to log distinct messages for Cosmos and DocumentDB.